join.py

```diff
@@ -0,0 +1,20 @@
+import itertools
+
+
+def join(*lists, sep='-'):
```

Comment:
Add return types and type hints - apply in all pull requests 
https://docs.python.org/3/library/typing.html
Read about it here